libenv/sysinfo.c

```diff
@@ -2666,103 +2666,119 @@ static void GetCPUInfo(EvalContext *ctx)
 #ifndef __MINGW32__
 
 /**
-   Return the number of seconds the system has been online given the current
-   time() as an argument, or return -1 if unavailable or unimplemented.
+   Return the current boot time (number of seconds since Epoch).
 */
-int GetUptimeSeconds(time_t now)
+time_t GetBootTime(void)
 {
-    time_t boot_time = 0;
+    static time_t boot_time = 0;
     errno = 0;
 
+    if (boot_time == 0)
+    {
 #if defined(BOOT_TIME_WITH_SYSINFO)         // Most GNU, Linux platforms
 
-    struct sysinfo s;
-    if (sysinfo(&s) == 0)
-    {
-       // Don't return yet, sanity checking below
-       boot_time = now - s.uptime;
-    }
+        time_t now = time(NULL);
```

Comment:
Aah, now I see what I mean, well spotted. You're right, that may be a problem. I wonder how expensive it is to find out uptime every time. Hmm, need to think about that one...